src/httpAgent.test.js

```diff
@@ -0,0 +1,78 @@
+import { AxiosMocker } from '../testUtils/axiosMocker';
+import * as utils from './utils';
+import { HttpAgent } from './httpAgent';
+import * as globals from './globals';
+
+describe('httpAgent', () => {
+  test('postSpans - simple flow', async () => {
+    const reqBody = 'abcdefg';
+    globals.TracerGlobals.setTracerInputs({ token: 't_xxx' });
+    const { url } = utils.getEdgeUrl();
+
+    await HttpAgent.postSpans(reqBody);
+
+    const requests = AxiosMocker.getAxiosMocker().history.post;
+    expect(requests.length).toEqual(1);
+    expect(requests[0].data).toEqual(reqBody);
+    expect(requests[0].headers).toEqual({
+      Accept: 'application/json, text/plain, */*',
+      'Content-Type': 'application/json',
+      Authorization: 't_xxx',
+      'User-Agent': '@lumigo/tracerMock$1.2.3',
+    });
+    expect(requests[0].timeout).toEqual(250);
+    expect(requests[0].baseURL).toEqual(url);
+  });
+
+  test('postSpans - reject 500', async () => {
+    const reqBody = 'abcdefg';
+    globals.TracerGlobals.setTracerInputs({ token: 't_xxx' });
+
+    AxiosMocker.getAxiosMocker()
+      .onPost()
+      .reply(500);
+
+    await HttpAgent.postSpans(reqBody);
+
+    //no Error throwed
+  });
+
+  test('postSpans - reject network error', async () => {
+    const reqBody = 'abcdefg';
+    globals.TracerGlobals.setTracerInputs({ token: 't_xxx' });
+
+    AxiosMocker.getAxiosMocker()
+      .onPost()
+      .networkError();
+
+    await HttpAgent.postSpans(reqBody);
+
+    //no Error throwed
+  });
+
+  test('postSpans - reject network error', async () => {
```

Comment:
Isn't this test the same as the test above?

Reply:
No, the test above testing the response code, this a client error

Reply:
![image](https://user-images.githubusercontent.com/49228804/86139400-5a6d7500-baf8-11ea-9c10-fcbca8fbe4af.png)
Both of them check that there isn't an exception and they have the same name.